### PR TITLE
PGI-19: fixes for cuda module

### PR DESCRIPTION
**Pull request by hoopoepg**
- set of changes for cuda module to support PGI 19

### CODE REVIEW

File: src/uct/cuda/base/cuda_md.c [hunk 1]
```diff
@@ -20,7 +20,7 @@ ucs_status_t uct_cuda_base_detect_memory_type(uct_md_h md, const void *addr,
                                               size_t length,
                                               ucs_memory_type_t *mem_type_p)
 {
-    CUmemorytype memType = 0;
+    CUmemorytype memType = (CUmemorytype)0;
```
Comment (brminich) [inline src/uct/cuda/base/cuda_md.c]: minor: can set it to `CU_MEMORYTYPE_HOST` by default

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 16-Dec-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/d19c2bf7d0094f73e7347b4c1dbad86761e2213f/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/c0fd8dd4bd8abe3f1615a908d08ff73734afe1ee/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/6a343d22e55edb77a81bbe0f1aca209099acc69c/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/eb0e39d9468ec45dda98f346fbed9058ee4bf7c1/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/7184fef4188225c7295ac5d5e5003abe1d93b2c9/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/4202ec0a9512e35afbaea9c2ddf0a02fd505bdca/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/0476b6b9a7bcacdb9ec357ae2ac6f1397b4fcf17/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/01349283c2891a711d4d10d1b125ecd40ad21079/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/ada7d43f935654cb42bda20ce4cbd4dfb07e4e4d/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/6a7e91614c95fcae8e1b8e78b7a0da616bb9b2a2/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/8e7f80a14887f043b846a665e519f3b4159bb798/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/c342cc8902b57606432314dca803a57749607857/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/d62a260ba56de2c1ff3b6955567859357b78ad95/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/4193f5059dd0051e50a469728304e89e0a2a27d9/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/e18d79fe5b0446eceff316cf2082b76842d4659d/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/9095136e85b7bdfe4a80b615bdaedbd1886d7cfd/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/8997824ab5023cc2b96c5a7ca8cc501a4ae54f21/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/4e0aef794e75b154a83d0fd0d458b28de2f0a121/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/454ba768a8155ce8d2e9075ebd85aa9db839280a/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/5ed0bbc38e19a37f4bd42896e6125c2081a3e9d5/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/22016ee90d643f65a54fa73839a98d0b74dd3c3c/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/bb7b61878b67f7c22f28db999f62d4f91555420c/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/a931e00bb2fe6962ea2f1b7cd5b7883295df230e/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/aa0b4c10c52affaf3ed6d4984d7f1097c34c4e86/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/85d4eba2759af05320c514ad8a7a8f61/raw/6e1eb1880bc59166d2a1777b93878f6f71f1aefe/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

File: src/uct/cuda/base/cuda_md.c (hunk 1 above)
Comment (hoopoepg) [reply src/uct/cuda/base/cuda_md.c]: yes, but we have add static assert for this case

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>FAILED</b> on 5 of 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 16-Dec-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/3942455666e5ba4807232f1cb806f9abbea2aa33/_main) | :x: FAILURE
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/82e9394a3cae7fa7756d71196c292b5115c5a709/r-vmb-ppc-jenkins_W0) | :x: FAILURE
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/317b54e25c95618e5bc2184d5276f43eb8c261ca/r-vmb-ppc-jenkins_W1) | :x: FAILURE
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/7a86cfc084f9c0e778e7b7ab6a9648b357409287/r-vmb-ppc-jenkins_W2) | :x: FAILURE
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/f39fe9a5d796aaaa2d3d9db567c0a8b212348f54/r-vmb-ppc-jenkins_W3) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/92d0bdda97428302fd68ed23a47c2599e5b08ab5/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/2ddf13354c31455ea9aed6ba0485ea7e7b8d7baf/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/5781fc79359fbe89c411e0e51c4831def3021586/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/0f102c4d2b93e1775a81f4a77d7cd19e81f3b92c/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/08f21d2528388a4594456c6da42b14e749598be1/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/672e931643114793ca895fd49224fe40a4c80cb5/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/dd7385a2f1aa7f9b73e2581acc3c4c53f61fd260/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/885e45d2a61f2036f2f4f675908dd8cade4fedb7/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/36d3e00770d455492be4a9352ff7dee2ee7d9ad0/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/5f58a834fd2be985024d907c7b05365333c7bde9/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/f839b68d155b4e10e49b38f3b150f1792decda41/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/31c217705e0d93f6f2a0f7c38faa3d63bfba3786/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/1bca1a778613643bca2a57446b81fa4a3023ccb7/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/2fa429b00de142dcaa7a8ed0e0967e066b43e00c/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/2bfed4bb3aea9e2edc282ce3eaa2cbb7f4ed965a/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/f28806bf222a42a550be480b84b197ca584043ca/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/5b1241d16dacaff3942b409830de648bf5e0950c/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/130a6d8f7f807eb6190ac6ede4e5b8aedae5c8fa/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/25a86c183c5b40ca299deafab552b9554d86717c/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/89ed41859fc8904c25cef6ba4ac1e0c6/raw/760972e17fae3bd42824af2826e41afd7fc74617/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS

**Comment by hoopoepg** [comment]
source not in tree
bot:mlx:retest

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 17-Dec-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/7e22e84795c3d4c5edbff51576713f692c18fc9d/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/48ffdb4ab5a4d42daa29ce25cd39933d4887626f/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/bd351ad98cd2d51b2fd30bf3a9f52610248dc34f/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/2ab16623952d4feda00cb86a7f247559557e7348/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/2f36a8eda4aa23117704db7ef2bde21979c24f59/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/231c302410a30b5c2a53a8eb8bd02ca486ddcf7f/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/eb582a607a3e81f6449152cdfdccda8e9c5270a1/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/4ee158ea5483395ca4c7fbecabf51ac7a6e62ca8/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/85b16da6f183855a84a0a547f7b082c12ba65e33/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/286fab1072693ac1ae7f565be5e9df9a5b29b6d1/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/d0939c1730cc35522d3561daa0a3512c044ee9cf/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/aa814a13e603c2532bbb71326be031a5dd25ac57/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/220daf98e21752a1717629ca640e66799b6d0518/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/ca5674df578dad16c6e1610b52be538986ce5f71/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/36626fd301bc73ed367d79938901ab771c47e785/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/e298dd9729f3cb754e48cfd8ddc26ef46349d9c3/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/f40e0eaf62b4316a3f353c1b762fe344cea519c0/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/367288a9e62bb8518f6df39567cad18f343a2c12/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/32d8cb2ffbdd070df7138b4c15eb6f7a5a8b695a/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/39beac67983281490e8adbf0af2cc91b4fce9775/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/e0ccb259a0bf3a8d039754547431bff3fbb2ca6d/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/ed19df63be2ae94010b0a345db930cf1e94dd97e/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/bc9058fe50a5ea22461a880e25f93f59ab460072/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/26910cdd78b7eed06cb89d5177d8777e35379326/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/14edd52833ce112010a74b7c14334db5/raw/0cad3f0ddbafa5bff55c4eb2fbdf0bcb1cc5d79d/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by hoopoepg** [comment]
@yosefe ok to merge?

**Pull request merged**